vision/api/VisionTest/ProductSearchTest.cs

```diff
@@ -0,0 +1,260 @@
+
+            output = _productSearch.Run("list_products", _projectId, REGION_NAME);
+            Assert.Contains(PRODUCT_ID, output.Stdout);
+            Assert.Contains(PRODUCT_ID_2, output.Stdout);
+        }
+
+        [Fact]
+        public void TestAddProductToSet()
+        {
+            _productSearch.Run("create_product", _projectId, REGION_NAME, PRODUCT_ID, PRODUCT_DISPLAY_NAME, PRODUCT_CATEGORY);
+            _productSearch.Run("create_product_set", _projectId, REGION_NAME, PRODUCT_SET_ID, PRODUCT_SET_DISPLAY_NAME);
+
+            var output = _productSearch.Run("list_products_in_set", _projectId, REGION_NAME, PRODUCT_SET_ID);
+            Assert.DoesNotContain(PRODUCT_ID, output.Stdout);
+
+            output = _productSearch.Run("add_product_to_set", _projectId, REGION_NAME, PRODUCT_ID, PRODUCT_SET_ID);
+            Assert.Equal(0, output.ExitCode);
+
+            output = _productSearch.Run("list_products_in_set", _projectId, REGION_NAME, PRODUCT_SET_ID);
+            Assert.Contains(PRODUCT_ID, output.Stdout);
+        }
+
+        [Fact]
+        public void TestRemoveProductFromSet()
+        {
+            _productSearch.Run("create_product", _projectId, REGION_NAME, PRODUCT_ID, PRODUCT_DISPLAY_NAME, PRODUCT_CATEGORY);
+            _productSearch.Run("create_product_set", _projectId, REGION_NAME, PRODUCT_SET_ID, PRODUCT_SET_DISPLAY_NAME);
+            _productSearch.Run("add_product_to_set", _projectId, REGION_NAME, PRODUCT_ID, PRODUCT_SET_ID);
+
+            var output = _productSearch.Run("list_products_in_set", _projectId, REGION_NAME, PRODUCT_SET_ID);
+            Assert.Contains(PRODUCT_ID, output.Stdout);
+
+            output = _productSearch.Run("remove_product_from_set", _projectId, REGION_NAME, PRODUCT_ID, PRODUCT_SET_ID);
+            Assert.Equal(0, output.ExitCode);
+
+            output = _productSearch.Run("list_products_in_set", _projectId, REGION_NAME, PRODUCT_SET_ID);
+            Assert.DoesNotContain(PRODUCT_ID, output.Stdout);
+        }
+
+        [Fact]
+        public void TestProductSearch()
+        {
+            _productSearch.Run("import_product_set", _projectId, REGION_NAME, CSV_GCS_URI);
+
+            var output = _productSearch.Run("get_similar_products", _projectId, REGION_NAME, PRODUCT_SET_ID, PRODUCT_CATEGORY, Path.Combine("data", IMAGE_URI_1), SEARCH_FILTER);
+            Assert.Equal(0, output.ExitCode);
+            Assert.Contains(PRODUCT_ID, output.Stdout);
+        }
+
+        [Fact]
+        public void TestProductSearchGcs()
+        {
+            _productSearch.Run("import_product_set", _projectId, REGION_NAME, CSV_GCS_URI);
+
+            var output = _productSearch.Run("get_similar_products_gcs", _projectId, REGION_NAME, PRODUCT_SET_ID, PRODUCT_CATEGORY, REF_IMAGE_GCS_URI, SEARCH_FILTER);
+            Assert.Equal(0, output.ExitCode);
+            Assert.Contains(PRODUCT_ID, output.Stdout);
+        }
+
+        public void CleanUp()
```

Comment:
Rename this `Dispose()` unless you're calling it from elsewhere

It's idiomatic for the setup code to go into the constructor and the teardown code to go into dispose

Reply:
Done.